TCPclient.cpp

```diff
@@ -3,36 +3,55 @@
 
 #include <sys/types.h>
 #include <sys/socket.h>
+#include <unistd.h> //included this for the close function
+#define _XOPEN_SOURCE_EXTENDED 1
 
-#include <netinet/in.h>  
+#include <netinet/in.h> //contains socket structs
 
-int main() {
+int main()
+{
 
- 	//creating client socket 
+	// creating client socket
 	int network_socket;
-	network_socket = socket(AF_INET, SOCK_STREAM, 0)
-
-	// specify an address for the socket 
-	struct sockaddr_in server_address; 
-	server_address.sin_family = AF_INET;    //what type of address?
-	server_address.sin_port = htons(9002); 
+	network_socket = socket(AF_INET, SOCK_STREAM, 0);
+
+	// specify an address for the socket to connect to
+	struct sockaddr_in server_address;
+	// need to define a few fields within the sockaddr_in struct
+	// what is the family
+	server_address.sin_family = AF_INET;
+	// specify port
+	// htons = host to network (making sure data format is correct)
+	// puts integer port into correct byte order
+	// pass the actual port, htons takes care of conversion
+	server_address.sin_port = htons(9002);
+
+	// specify actual address
+	// server_address is a struct with a sin_addr struct member which contains
+	// s_addr field
+	// conecting to loval machine, can use any address connected to local machine
 	server_address.sin_addr.s_addr = INADDR_ANY;
 
-	int connection_status = connect(network_socket, (struct sockaddr *) &server_address, sizeof(server_address));
+	// first arg is actual socket
+	// 2nd param, need to cast pointer  to struct sockaddr
+	// 3rd param is size of address
+	int connection_status = connect(network_socket, (struct sockaddr *)&server_address, sizeof(server_address));
 
-	//check for error with connection 
-	if (connection_status == -1) {
+	// check for error with connection. if connect = -1, means error connecting
+	if (connection_status == -1)
+	{
 		printf("There was an error making a connection to the remote socket\n\n");
 	}
-	//recieve data from the server
-	har server_response[256]
+
+	// recieve data from the server using recv function
+	char server_response[256]; // hold message that we get back from server
```

Comment:
Probably not great practice.  You could use dynamic memory or file scope memory.  This might be ok.  Try some other way to do this?  What happens if you UDP packet response is larger than 256 bytes?